scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+				else if (textRegionLines.size() == 1) 
+				{
+					pass = FrameworkLineTests::NEWLINE;
+#ifdef DEBUG_TEXT_IMPORT
+					qDebug() << "NEWLINE2 point:" << point << " _lastXY:" << lastXY << " origin: " << textRegioBasenOrigin << " this:" << this << " linespacing: " << lineSpacing;
+#endif
+				}
+			}
+		}
+		else
+		{
+#ifdef DEBUG_TEXT_IMPORT
+			qDebug() << "NEWLINE2 oops:"<<point << ":"<<textRegioBasenOrigin << ":" << lineSpacing;
+#endif
+		}
+	}
+	return pass;
+}
+
+// Just perform some basic checks to see if newPoint can reasonably be asscribed to the current textframe.
+TextRegion::FrameworkLineTests TextRegion::isRegionConcurrent(QPointF newPoint)
+{	
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+	}
+
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		xInLimits = true;
+	}
+	else
+	{
+		qDebug() << "X out of limits " << lastXY << " : " << newPoint;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);
+	return pass;
+}
+
+
+TextRegion::FrameworkLineTests TextRegion::moveToPoint(QPointF newPoint)
+{
+	//
+	//qDebug() << "moveToPoint: " << newPoint;
+	// Do some initilization if we are in a new text region
+	// we could also update these if glyphindex = -1;
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+		//qDebug() << "newPoint3: " << newPoint;
+	}
+	
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	//TODO: x limiting should be different for moving as opposed to adding a new glyph because moiving is due to a discontinuity in glyphs
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		//qDebug() << "newPoint6: " << newPoint;
+		xInLimits = true;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		//qDebug() << "newPoint7: " << newPoint;
+		yInLimits = true;
+	}
+	//qDebug() << "newPoint8: " << newPoint;
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);
+	//TODO: need to check to see if we are creating a new paragraph or not. basically if the cursor is returned to x origin before it reached x width. this could be returned as part of a matrix by linearTest that specifies exactly how the test ws passed. maybew return an enum with either the mode that passed or a failure value
+	if (pass != FrameworkLineTests::FAIL)
+	{
+		//create new lines and line segments depending upon the moide of the movement.
+		TextRegionLine *textRegionLine = nullptr;
+
+		if (pass == FrameworkLineTests::NEWLINE || pass == FrameworkLineTests::FIRSTPOINT)
+		{
+			//qDebug() << "Newline: ";
+			if (pass == FrameworkLineTests::FIRSTPOINT)

```

Comment:
it's imo very dangerous, but jean will not want to see `{`s here, since the if are single commands: 

```
if blah
  if buh
    do
else
  other
```

Reply:
wait, no, both branches do the same:

```
if (pass != FrameworkLineTests::FIRSTPOINT) || textRegionLines.empty())
  textRegionLines.push_back(TextRegionLine());
```

Reply:
that's neeter